### DISCUSSION TITLE
Is the okd project still on track?

### DISCUSSION THREAD

**Discussion by titou10titou10** [okd-project/okd#1922] [General]
Hi everybody,

The goal of this discussion is to get some news from the OKD project and its current state in a constructive way

I'm asking this because since weeks, or even since months for some flavours, nightly build are failing as you can see here https://amd64.origin.releases.ci.openshift.org/

- OKD/FCOS v4.15 builds are failing as for v4.16, v4.14, v4.13.. For v4.16 it is understandable as it is still alpha and the other older versions are obsolete....
- Same for OKD with SCOS, v4.15 and also v4.16, v4.14 and 4.13... are failing since weeks/months...

Q:
- what's the current state of the project?
- is there someone that monitor/maintain/fix the OKD code/nighly builds/releases?
- when can we expect a new working version  for OKD/FCOS and OKD/SCOS (v4.15 or v4.16)?
- is help from the community needed for this project? if so in what area?

Thanks

**Comment by JaimeMagiera** [comment]
Hello,

Thanks for reaching out. The project is in transition while the various participants (Red Hat, Working Group, etc.) work towards addressing some long-standing issues, while at the same time providing greater transparency and participation. There are no planned builds or bug fixes using the current infrastructure and processes. We are working diligently to define next steps on new infrastructure, with new processes. The Working Group will be releasing a statement shortly. 

The Working Group can certainly use more help. Right now, while the build details are hashed out, we need folks willing to improve our existing documentation and our website by identifying problems. Is this something you could help with?

**Comment by titou10titou10** [comment]
Thanks to give us some feedback and good to read that something is coming to continue to run this project and make if transparent and participative
Sure for the doc review. How? by visiting the web pages? how to report? create issues? could be tedious

**Comment by madpearl** [comment]
Hello,
 is there news about further going/developing of the OKD Project?
There is no new videos in "OKD | Working Group"   playlist in YouTube, the last one was on 13.02.2024. I sow in slack channels, that there is meetings after the last video, but not published. Please, give us Info!
Regards,
Bisser Danev

**Comment by marcusaram** [comment]
@JaimeMagiera Thank you for your response. Any idea when soon is? And could you explain what the consequences are if these changes are done? As far as I know (based on very limited information) is that the infrastructure and future containers will be RHEL free (also the UBI) and need to use CentOS. As I can understand the first step, the second step (by not using UBI for upstream) will probably cause some nice side effects as bugs that do appear on OKD and not OCP and vice versa.

It would really help to get a statement out what the future will be, also I (and probably more) are willing to help in whatever is needed.

**Reply by JaimeMagiera** [reply]
Hi. The statement should be done in the next week or so. We were waiting on confirmation from Red Hat on their commitment and waiting to see how things went with initial container builds. The news is good on all fronts. Stay tuned. Thank you for your patience.

**Comment by titou10titou10** [comment]
Any progress on this matter?

**Comment by JaimeMagiera** [comment]
@titou10titou10 The statement is done and will be posted to the website and social media today. When it's up, I'll provide the link here. We're also going to use this thread to gather further question to include in the FAQ page we're setting up. Stay tuned.

**Reply by nate-duke** [reply]
is "the website" the [okd blog](https://www.okd.io/blog/) or somewhere more IBM related?

**Reply by adarshaj** [reply]
> is "the website" the [okd blog](https://www.okd.io/blog/) or somewhere more IBM related?

The post is now live at the blog link you mentioned, the post link is https://www.okd.io/blog/2024/06/01/okd_future_statement/

**Comment by adarshaj** [comment]
So can we upgrade from OKD/FCOS to OKD/SCOS without causing any applications to break?

Is there any significant change in terms of networking or underlying configurations (kernel command line params, dhcp/dns related changes)?

Will FCOS be supported back in future sometime?

Also, in the larger picture, OKD was supposed to be upstream for Openshift, is that still the case?

**Reply by marcusaram** [reply]
If I understand correctly.
FCOS and SCOS was until now only the OS for the nodes. Until now all the container images were build on UBI (RHEL based). That is going to change to SCOS. So for the nodes there will probably still FCOS and SCOS choices, but the containers will use different base image. I understand this, but it feels like corporate driven (IBM). Red Hat should absolutely know that OKD was build on UBI

**Reply by darren-oxford** [reply]
Just my personal opinion, but I think SCOS for the nodes is the worst of both worlds, it is a moving target just like FCOS, but brings with it an old kernel and libraries with backported fixes. As the kernel ABI/API is stable, we don't really care about the underlying kernel version.

Having an old kernel and libraries makes some sense when you have a huge collection of packages in the distribution repo, to ensure that the distro has a known base. Our aim for OKD, is to run a paired down OS on the nodes, with just enough to support the hardware and start the Kublet. It makes very little sense holding onto an old kernel when running containers as we include any library dependencies in the container anyway. FCOS at least has all the latest hardware support and actually in my experience performs better as much as 15% for some workloads due to tweaks in the newer kernel.

SCOS seems to have more benefits for RedHat than it does the community, it allows RedHat to try new things out that they want to put into the next iteration of RCOS, but it is not that far ahead of RCOS to offer much in the way of benefits to the community. For me there is also a bit of a concern about SCOS being based on CentOS and the way that project is managed and I have reservations about how the community might be further closed out of CentOS, for that reason I prefer the more community focused approach of Fedora.

Note: I am not opposed to RedHat trying to monitize their work, or their attempts to make it more difficult for other commercial organisations to 'freeload', I am just concerned about how those efforts have affected various community efforts.

**Reply by marcusaram** [reply]
I also tested FCOS vs SCOS. My experience with FCOS and OKD were time to time issues, it feels that FCOS was (indeed) too bleeding edge compared to the rest of the components. We didn't experience any notable performance difference, but we don't really have any CPU or IO bound workload. Newer kernels should get more performance out our CPUs (EPYC Genoa).

I personally think that the current choice of running FCOS or SCOS is a good one, so you can choose what you prefer.

I really miss the information what is/will be changed, are this only build servers or are the components in OKD? Moving away from UBI for OKD it will be really challenging.

**Reply by ausil** [reply]
As the OKD builds were an artifact of OCP CI, when OCP switched to using ELS containers from UBI, so did OKD. This is where things got tricky.  Moving forward, the containers will be based on CentOS stream, and we are setting OKD up as the upstream for OCP. 

There is no reason not to offer both OS's in the future; for now, having only SCOS is a path to help move everything into the community space, where people interested can work on different options. I really hope that interested people will work on offering FCOS down the road.

**Reply by marcusaram** [reply]
Is there any reason to switch to ELS containers? So the containers need to update to the current UBI version will also fix the problem? Switching to SCOS mean even more 'bleeding-edge' than UBI (and way more than ELS)? Correct me if I'm wrong.

Is it possible to elaborate more what is happening? And why these choices are made (incl. some history)?

**Reply by darren-oxford** [reply]
@marcusaram There seems to be a little confusion here. As I understand it...

- OKD is currently offered with a choice of FCOS and SCOS as the OS for the nodes.
- OKD is currently built from OpenShift sources using the RedHat prow build system.

The breaking change is that OpenShift now includes [some] ELS containers in their build system, resulting in OKD including those ELS images in the OKD build as well. RedHat have probably started using ELS images simply because they have images based on EL7 or 8 that rather than rebuild for EL9 they want to continue to use the older image. Not wishing to be cynical, it may simply be lack of available time to update them all, or an attempt at maintaining consistency for users. Either way OKD cannot use them, unfortunately another change made by RedHat that has [unintended] consequences for the community project.

The solution being proposed is to move from the prow build to a Tekton pipline, this is something that has been in the works for a while. It seems the plan for now is to concentrate on a single build due to the complexity of building for two OS's. The choice of SCOS for the node OS, is likely the (IMHO misguided) expectation that building for SCOS will be easier than FCOS due to fewer changes, combined with the benefits that SCOS has to RedHat as a testing ground for the next RCOS iteration. Given that RedHatter's make up the bulk of those working on OKD, that is not an entirely unreasonable objective for them. The same applies  substituting ELS with CentOS stream for container images. It would however, be good to know how these choices were made as opposed to keeping the FCOS base and substituting ELS for UBI for example.

Incidentally, it should be said that a straight substitute of either CentOS or UBI may not be as easy as it sounds, there are likely reasons why RedHat have continued to use an old container image, that we the community will probably discover when trying to build them.

My hope is that once a working independent build system is established, that we will see greater community involvement and that OKD will become more like Fedora with community contributions rather than what it is now, essentially a vanilla re-build of OpenShift. It has been difficult for individuals to contribute to OKD using the current build system, hopefully that should change now.

**Comment by titou10titou10** [comment]
Some questions about installation.
Will the "new" OKD
... be available on the same platforms for installing as before: bare, AWS, Azure...
... propose the same install methods as for now: UPI/IPI, agent installer, assisted installer...
... have the same "update" channels, ie only a "stable" channels vs  3 for OCP
... still be be associated to insight and the clusters visible online via the RH "hybrid console" ?..and the "evaluation period" for OKD fixed
... require a red-hat account to get the extensive components of OCP?

**Reply by GingerGeek** [reply]
Thanks for taking the time - we discussed your questions at the working group meeting last night.

**... be available on the same platforms for installing as before: bare, AWS, Azure...**
Yes, the installer will be functionally the same and we expect to support the same installation methods. I understand that RH are in the process of ensuring that CentOS Stream CoreOS (SCOS) has downloadable artifacts for all the major cloud providers similar to [FCOS](https://fedoraproject.org/coreos/download?stream=stable#arches) as currently you need to boot FCOS first and then rpm-ostree switches it over to SCOS. I lost the JIRA ticket for it but will link once I find it again

**... propose the same install methods as for now: UPI/IPI, agent installer, assisted installer...**
Yes, the installer will be functionally the same and we expect to support the same installation methods. Personally, I only have experience with bare metal UPI but community members have had success with IPI and agent-based - sometimes work-arounds are needed. I think the CI for a release goes through IPI on a few cloud providers.

The RedHat hosted version of assisted installer doesn't support OKD afaik... would be nice to have an OKD version of it but there's probably quite a bit of work required there!


**... have the same "update" channels, ie only a "stable" channels vs 3 for OCP...**
Yes, OKD will continue to be single-channel for releases. We have no current plans (nor the resources) to support multiple channel releases.

**... still be be associated to insight and the clusters visible online via the RH "hybrid console" ?..and the "evaluation period" for OKD fixed**
No (well maybe), this is not intended behaviour. I'm not aware of any changes being done that will actively stop OKD registering into RHHC but OKD clusters are not supported by RedHat. The Hybrid Cloud console is opensource and you could I guess host your own one but my understanding is that it's an unintended consequence that you can register OKD into the openshift HC console.


**... require a red-hat account to get the extensive components of OCP?**
No, we have a project [OKDerators](https://github.com/okd-project/okderators-catalog-index/) that is established to try and bring the extensive catalog of operators (ODF, Virtualisation, ACM, AKS, etc) into OKD. All the code is open source it just(!) requires build plumbing to get it into a catalog that is consumable by OKD clusters.

**Reply by GingerGeek** [reply]
The JIRA link is: https://issues.redhat.com/browse/COS-2797

**Reply by rbaumgar** [reply]
@titou10titou10 when using a Red Hat account (pull secret) in your OKD cluster you need an OpenShift subscription because you are using binaries from OpenShift that are not part of OKD. e.g. using any OpenShift operators...

**Comment by titou10titou10** [comment]
Any timeframe when the nightly/automatic build will be available? as for when the first "stable" version will be ready to be deployed of OKD-SCOS?
we are particularly interested in the "bare/UPI" and "bare/UPI/agent installer" install methods. 
"bare/UPI/agent installer" was working fine for OKD-FCOS but I've never been able to deploy OKD-SCOS with this method..

**Reply by Goose29** [reply]
UPI installation method for SCOS is possible without issue, I have tried a couple different versions with FCOS as the starting point, and then rpm-ostree'ing into SCOS (since the OS binaries for SCOS weren't... and still isn't available?).

**Comment by blackcobra1973** [comment]
ok, does this means I have to do a complete reinstall from FCOS to CSOS for the new updates or that will not be any issue?   Or and upgrade path from FCOS to CSOS for the nodes is made available in the update ?

**Reply by msteenhu** [reply]
I have been told on Slack (kubernetes/okd-dev) it is the goal to backport the changes to 4.16. In theory it is then possible to do a (forced) upgrade from the old FHCOS to the newly built SCOS.. Fingers crossed that the community can pull it off. I am already preparing for some sweaty upgrade maintenance windows..

**Reply by Goose29** [reply]
Nothing stops you from moving from a existing FCOS installation to SCOS, refer to the following [slack thread](https://kubernetes.slack.com/archives/C6AD6JM17/p1698147417515049). But I would highly recommend trying the upgrade path from FCOS to SCOS in a test cluster before going all out since some versions as mentioned in the threads aren't 100% compatible.

**Reply by Tyrone-W** [reply]
I've upgraded a test cluster from FCOS 4.12 to latest FCOS 4.15 then upgraded it to a nightly SCOS 4.16(purely to test), this currently requires a manual intervention shown in https://github.com/okd-project/okd/discussions/1971#discussioncomment-10119718. 
In the most recent community meeting this was confirmed as a expected step for FCOS to SCOS migration of OKD.

**Reply by nfkamdhhc** [reply]
> Nothing stops you from moving from a existing FCOS installation to SCOS, refer to the following [slack thread](https://kubernetes.slack.com/archives/C6AD6JM17/p1698147417515049). But I would highly recommend trying the upgrade path from FCOS to SCOS in a test cluster before going all out since some versions as mentioned in the threads aren't 100% compatible.

Well, SCOS has many compatibility issues with older server hardware, that is the main reason I chose FCOS as base. I find it strange, that they speak so much of community, yet choose the OS that is less compatible with older hardware, as far as my experience goes home lab servers are usually older, thus having the support dropped is pain. But we will see how the chips land.

**Comment by MaAb2401** [comment]
Hi, 
There is link to download SCOS in ova format ?

Thanks

**Reply by nate-duke** [reply]
or a VHDX for those of us that are Broadcom refugees!

**Reply by joelcapitao** [reply]
The images are hosted in https://cloud.centos.org/centos/scos/ but apparently it's not up-to-date, seems like the syncing is broken. 
Though, those images are not built in the CentOS Infra, but in Red Hat CoreOS one, and synced to cloud.centos.org.
The source lives in https://github.com/coreos/rhel-coreos-config

**Reply by joelcapitao** [reply]
So, the syncing is not broken. The images are not updated that often as it's just used as bootimage for OKD. Once boot in OKD, the node pivots to newer image with fresh content.

**Reply by MaAb2401** [reply]
> So, the syncing is not broken. The images are not updated that often as it's just used as bootimage for OKD. Once boot in OKD, the node pivots to newer image with fresh content.

What do you mean by "the node pivots to newer image with fresh content"? Are you referring to the system update that can be done after the OKD cluster deployment?

**Reply by joelcapitao** [reply]
Yes, with the rpm-ostree rebasing processus i.e: `rpm-ostree rebase`
